lib/registration.pm

```diff
@@ -229,17 +229,23 @@ Wrapper for SUSEConnect -p $name  over ssh.
 =cut
 
 sub ssh_add_suseconnect_product {
-    my ($remote, $name, $version, $arch, $params, $timeout, $retries, $delay) = @_;
+    my ($remote, $name, %args) = @_;
     assert_script_run "sftp $remote:/etc/os-release /tmp/os-release";
     assert_script_run 'source /tmp/os-release';
-    $version //= '${VERSION_ID}';
-    $arch //= '${CPU}';
-    $params //= '';
-    $timeout //= 300;
-    $retries //= 3;
-    $delay //= 10;
-
-    script_retry("ssh $remote sudo SUSEConnect $debug_flag -p $name/$version/$arch $params", delay => $delay, retry => $retries, timeout => $timeout);
+    $args{program} //= 'SUSEConnect';
+    $args{version} //= '${VERSION_ID}';
+    $args{arch} //= '${CPU}';
+    $args{params} //= '';
+    $args{timeout} //= 300;
+    $args{retries} //= 3;
+    $args{delay} //= 10;
+
+    record_info('REG_DBG', $remote . ' ' . $name . ' ' . $args{program} . ' ' . $args{version} . ' ' . $args{arch} . ' ' . $args{params} . ' ');
+    if ($args{program} eq 'registercloudguest') {
```

Comment:
Maybe add a comment explaining how if and else are different. How they are?

Also consider a pattern like this when you have to compose commands with many arguments
```
script_retry(join(' ', 'ssh',  $remote, 'sudo',  $args{program}, $args{params}),
    delay => $args{delay},
    retry => $args{retries},
    timeout => $args{timeout});
```

Reply:
I used `sprintf` please recheck